src/Sarif/Writers/ConsoleLogger.cs

```diff
@@ -15,7 +15,7 @@ public class ConsoleLogger : BaseLogger, IAnalysisLogger
         //  TODO:  We directly instantiate this logger in two classes, creating 
         //  unamanged dependencies.  Fix this pattern with dependency injection or a factory.
         //  #2272 https://github.com/microsoft/sarif-sdk/issues/2272
-        public ConsoleLogger(bool quietConsole, string toolName, IImmutableSet<FailureLevel> levels = null, IImmutableSet<ResultKind> kinds = null) : base(levels, kinds)
```

Comment:
>IImmutableSet [](http://example.com/codeflow?start=65&length=13)

immutable set can't easily be serialized/deserialized, so we create alias types to these constructs.